src/plugins/intel_cpu/src/gemm/ov_cpu_gemm.cpp

```diff
@@ -0,0 +1,122 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "ov_cpu_gemm.h"
+
+#include <string>
+#include <vector>
+
+#include "openvino/core/parallel.hpp"
+#include "onednn/dnnl.h"
+#include "mlas.h"
+
+namespace ov {
+namespace cpu {
+class ThreadPool {
+public:
+    ThreadPool() = delete;
+    explicit ThreadPool(const size_t& threadNum) : threadNum(threadNum) {}
+public:
+    // the actual threads used for sgemm
+    size_t threadNum = 0;
+};
+size_t DegreeOfParallelism(ThreadPool* tp) {
+    // threadpool nullptr means single threaded
+    return tp ? tp->threadNum : 1;
+}
+void TrySimpleParallelFor(ThreadPool* tp, const std::ptrdiff_t total, const std::function<void(std::ptrdiff_t)>& fn) {
+    if (tp == nullptr) {
+        for (std::ptrdiff_t i = 0; i < total; i++) {
+            fn(i);
+        }
+    } else {
+        ov::parallel_nt(tp->threadNum, [&](const size_t ithr, const size_t nthr) {
+            std::ptrdiff_t start = 0, end = 0;
+            ov::splitter(total, nthr, ithr, start, end);
+            for (std::ptrdiff_t i = start; i < end; i++) {
+                fn(i);
+            }
+        });
+    }
+}
+size_t getCacheSize(int level, bool perCore) {
+    return dnnl::utils::get_cache_size(level, perCore);
+}
+};  // namespace cpu
+};  // namespace ov
+
+size_t ov_sgemm_pack_get_size(const int64_t N, const int64_t K) {
+    return MlasGemmPackBSize(N, K);
+}
+
+void ov_sgemm_pack(const char* transb,
+                   const int64_t N,
+                   const int64_t K,
+                   const int64_t ldb,
+                   const float* src,
+                   float* dst) {
+    MlasGemmPackB(*transb == 'T' ? CblasTrans : CblasNoTrans, N, K, src, ldb, dst);
+}
+
+void ov_sgemm(const char* transa,
```

Comment:
This is not ov_sgemm. This is exactly mlas_sgemm. Please see my comment above. From my understanding we should place mlas impl inside corresponding folder. 
src/sgemm implementation should serve as dispatcher between different sgemm providers. Basic impl can look like that:
```
void sgemm(...) {
#if defined (OV_CPU_WITH_MLAS)
  mlas_sgemm(...);
#else 
  THROW(); // or fallback on oneDNN/ngraph reference
#endif
}
```

However given the real dispatching is done on FC level we might even avoid introduction of this common interface for now